src/kaboom.ts

```diff
@@ -5467,11 +5477,335 @@ export default (gopt: KaboomOpt = {}): KaboomCtx => {
 
 	}
 
-	function obstacle(): ObstacleComp {
+	function obstacle(isObstacle: boolean = true): ObstacleComp {
+		let _isObstacle = isObstacle
 		return {
 			id: "obstacle",
 			require: [ "tile" ],
-			isObstacle: true,
+			setObstacle(this: GameObj<ObstacleComp | TileComp>, isObstacle: boolean) {
+				if (_isObstacle !== isObstacle) {
+					_isObstacle = isObstacle
+					this.getLevel().invalidateNavigationMap()
+				}
+			},
```

Comment:
Is it possible to use getter / setter here?

Reply:
Changed it. Didn't know we used get/set from reading the Kaboom2000 code. I like get/set better for most things.